marketplace/src/benchmarking.rs

```diff
@@ -17,236 +17,166 @@
-		drop(Marketplace::<T>::add_account_to_allow_list(alice.clone().into(), mkp_id, bob_lookup.clone()));
-
-	}: _(alice.clone(), mkp_id, bob_lookup)
-	verify {
-		let allow_list: AccountVec<T> = bounded_vec![];
-		assert_eq!(Marketplaces::<T>::get(mkp_id).unwrap().allow_list, allow_list);
+		let marketplace_id = Marketplace::<T>::get_next_marketplace_id() - 1;
+		assert_eq!(Marketplaces::<T>::contains_key(marketplace_id), true);
+		assert_eq!(Marketplaces::<T>::get(marketplace_id).unwrap().owner, alice);
 	}
 
 	set_marketplace_owner {
-		let (mkp_id, ..) = prepare_benchmarks::<T>();
-
+		prepare_benchmarks::<T>();
+		let alice_origin = origin::<T>("ALICE");
 		let bob: T::AccountId = get_account::<T>("BOB");
 		let bob_lookup: <T::Lookup as StaticLookup>::Source = T::Lookup::unlookup(bob.clone());
-
-	}: _(get_origin::<T>("ALICE"), mkp_id, bob_lookup)
+		let marketplace_id = Marketplace::<T>::get_next_marketplace_id() - 1;
+	}: _(alice_origin, marketplace_id, bob_lookup)
 	verify {
-		assert_eq!(Marketplaces::<T>::get(mkp_id).unwrap().owner, bob);
+		assert_eq!(Marketplaces::<T>::get(marketplace_id).unwrap().owner, bob);
 	}
 
-	set_marketplace_type {
-		let (mkp_id, ..) = prepare_benchmarks::<T>();
-
-	}: _(get_origin::<T>("ALICE"), mkp_id, MarketplaceType::Private)
+	set_marketplace_kind {
+		prepare_benchmarks::<T>();
+		let alice_origin = origin::<T>("ALICE");
+		let marketplace_id = Marketplace::<T>::get_next_marketplace_id() - 1;
+	}: _(alice_origin, marketplace_id, MarketplaceType::Private)
 	verify {
-		assert_eq!(Marketplaces::<T>::get(mkp_id).unwrap().kind, MarketplaceType::Private);
+		assert_eq!(Marketplaces::<T>::get(marketplace_id).unwrap().kind, MarketplaceType::Private);
 	}
 
-	set_marketplace_name {
-		let (mkp_id, ..) = prepare_benchmarks::<T>();
-
-		let new_name: NameVec<T> = bounded_vec![40, 30, 20];
-	}: _(get_origin::<T>("ALICE"), mkp_id, new_name.clone())
+	set_marketplace_configuration{
+		prepare_benchmarks::<T>();
+		let alice: T::AccountId = get_account::<T>("ALICE");
+		let alice_origin = origin::<T>("ALICE");
+		let marketplace_id = Marketplace::<T>::get_next_marketplace_id() - 1;
+		let marketplace_offchain_data =
+			BoundedVec::try_from(vec![1; T::OffchainDataLimit::get() as usize])
+				.expect("It will never happen.");
+		let marketplace_account_list: BoundedVec<T::AccountId, T::AccountSizeLimit> =
+			BoundedVec::try_from(vec![alice.clone(); (T::AccountSizeLimit::get() / 100) as usize]).unwrap();
+	}: _(alice_origin, marketplace_id, ConfigOp::Set(MarketplaceFee::Percentage(PERCENT_50)), ConfigOp::Set(MarketplaceFee::Percentage(PERCENT_50)), ConfigOp::Set(marketplace_account_list.clone()), ConfigOp::Set(marketplace_offchain_data.clone()))
 	verify {
-		assert_eq!(Marketplaces::<T>::get(mkp_id).unwrap().name, new_name);
+		let marketplace = Marketplaces::<T>::get(marketplace_id).unwrap();
+		assert_eq!(marketplace.commission_fee, Some(MarketplaceFee::Percentage(PERCENT_50)));
```

Comment:
Instead of checking if the fields are correct,  I would suggest you to build a new marketplace object with the expected data and then compare the existing marketplace object with the expected one.